pyk/src/pyk/kast/outer_lexer.py

```diff
@@ -190,23 +191,29 @@ class State(Enum):
 
 
 class LocationIterator(Iterator[str]):
-    _line: int = 1
-    _col: int = 0
+    _line: int
+    _col: int
     _iter: Iterator[str]
+    _nextline: bool
 
-    def __init__(self, x: Iterable[str]) -> None:
+    def __init__(self, x: Iterable[str], line: int = 1, col: int = 0) -> None:
         self._iter = iter(x)
+        self._line = line
+        self._col = col
+        self._nextline = False
 
     def __next__(self) -> str:
         la = next(self._iter)
         self._col += 1
-        if la == '\n':
+        if self._nextline:
             self._line += 1
-            self._col = 0
+            self._col = 1
+        self._nextline = la == '\n'
         return la
 
     @property
     def loc(self) -> Loc:
+        """Returns the line,column of the last character returned by the iterator"""
```

Comment:
What if no characters have been returned yet?

Reply:
You'll get `(1,0)`. There's no use case for getting the location before reading anything, so I didn't really think about it here. This is the only instance where you would get a location with a column of `0`, though, so that could work out as an indicator that no characters have been returned so far.

Reply:
Please indicate this in the docstring.